src/js/models/InfoViewModel.js

```diff
@@ -25,13 +27,19 @@ InfoViewModel.prototype.applyBindings = function() {
 InfoViewModel.prototype.hashChange = function() {
     if (location.hash === '#/info') {
         this.applyBindings();
+        this.infoText('Hide Info');
+    } else {
+        this.infoText('Show Info');
     }
 };
 
 InfoViewModel.prototype.toggleInfo = function() {
     if (location.hash === '#/info') {
         this.infoText('Show Info');
-        history.go(-1);
+        if (history.length > 2)
+            history.back();
+        else
+            location = '#';
```

Comment:
We should use `window.location` instead of `location`. Just noticed this because JSHint says `Error: Read only` for this line.

Also, http://stackoverflow.com/questions/4709037/window-location-versus-just-location